torchsde/_core/sdeint.py

```diff
@@ -193,21 +193,23 @@ def check_contract(sde, method, logqp, ts, y0, bm, names, adjoint_method=None):
             if diffusion_shape_[-1] != noise_channels:
                 raise ValueError("Every Tensor return by the diffusion must have the same number of noise channels.")
 
+    if not torch.is_tensor(ts):
+        if not (isinstance(ts, tuple) or isinstance(ts, list)) or not all(isinstance(t, (float, int)) for t in ts):
+            raise ValueError(f"Evaluation times `ts` must be a 1-D Tensor or list/tuple of floats.")
+        ts = torch.tensor(ts, dtype=y0[0].dtype, device=y0[0].device)
```

Comment:
I think it's BTAFTP here and just do:
```
if not torch.is_tensor(ts):
    ts = torch.tensor(ts, dtype=, device=)
```
If you disagree then I'd change `isinstance(ts, tuple) or isinstance(ts, list)` to `isinstance(ts, (tuple, list))`.

Additionally (outside the if statement) also check that it's one dimensional and monotonic. (And floating point as well if we want to be really careful.)

Reply:
Done. 

The mono check is in `integrate`. This also ensures that the adjoint solve is checked, which I think is great.